mne/annotations.py

```diff
@@ -395,6 +418,10 @@ def append(self, onset, duration, description, ch_names=None):
             Description for the annotation. To reject epochs, use description
             starting with keyword 'bad'.
         %(ch_names_annot)s
+        metadata : dict | array-like
+            Metadata for the annotation. Can be a dict or an array-like
+            object of dicts. If an array-like object, must be the same length
+            as ``onset``.
```

Comment:
with this we'd be exposing `metadata` to the user --> weren't we discussing keeping it private, and having dedicated functions populate / add / use / change this data?

If we make it public here (exposing it in the docstring), then I think we might as well make it completely public :thinking: